core/src/main/scala/org/apache/spark/shuffle/BlockStoreShuffleReader.scala

```diff
@@ -41,6 +43,20 @@ private[spark] class BlockStoreShuffleReader[K, C](
 
   private val dep = handle.dependency
 
+  private def fetchContinuousBlocksInBatch: Boolean = {
+    val conf = SparkEnv.get.conf
+    val compressed = conf.get(config.SHUFFLE_COMPRESS)
+    val featureEnabled = conf.get(config.SHUFFLE_FETCH_CONTINUOUS_BLOCKS_IN_BATCH)
+    val serializerRelocatable = dep.serializer.supportsRelocationOfSerializedObjects
+    // The batch fetching feature only works for reading consolidate file written by
+    // SortShuffleWriter or UnsafeShuffleWriter.
+    val readConsolidateFile = !handle.isInstanceOf[BypassMergeSortShuffleHandle[_, _]]
```

Comment:
I'm not sure I agree with this. `BypassMergeSortShuffleHandle` writes to many files, but it will concatenate them to a single consolidate file at the end.

Reply:
This flag added in 407b1e0 for fixing the Jenkins UT `ReduceNumShufflePartitionsSuite` and `AdaptiveQueryExecSuite` failure. From my first round debugging, I found the output written by BypassMergeSortShuffleWriter can't read continuously as expected, let me check.

Reply:
Fix in c3b009f, thanks for the comment and guidance, the root cause is missing to register the new block id in serializer manager.